tests/models/edx/test_base.py

```diff
@@ -1,84 +1,24 @@
-        {"context__user_id": 213456},
-        {"context__org_id": "", "context__course_id": ""},
-        {"context__org_id": "org", "context__course_id": "course-v1:org+any+any"},
-    ],
-)
-def test_models_edx_base_event_with_valid_content(kwargs):
-    """Tests that a valid base event does not raise a ValidationError."""
-
-    try:
-        BaseEventFactory(**kwargs)
-    except ValidationError:
-        pytest.fail(f"Valid base event with {kwargs} should not raise exceptions")
 
-
-@pytest.mark.parametrize(
-    "kwargs,error",
-    [
-        ({"ip": "invalid_ip"}, "not a valid IPv4 address"),
-        ({"ip": "  "}, "not a valid IPv4 address"),
-        ({"event_source": "not_server"}, "unexpected value; permitted: 'server'"),
-        ({"page": "not_None"}, "value is not None"),
-        ({"page": 0}, "value is not None"),
-        ({"page": ""}, "value is not None"),
-        ({"page": {}}, "value is not None"),
-        ({"referer": 1}, "invalid or missing URL scheme"),
-        ({"referer": "this/is/not/a/valid/url"}, "invalid or missing URL scheme"),
-        ({"context__user_id": "invalid"}, "value is not a valid integer"),
-        ({"context__user_id": {}}, "value is not a valid integer"),
-        ({"context__org_id": None}, "org_id\n  none is not an allowed value"),
-        ({"context__org_id": {}}, "org_id\n  str type expected"),
-        ({"context__course_id": "+org+any"}, "course_id must match regex"),
-        (
-            {"context__course_id": "course-v1:+course+not_empty"},
-            "course_id must match regex",
-        ),
-        (
-            {"context__course_id": "course-v1:NOT_org+course+session"},
-            "course_id must match regex",
-        ),
-        (
-            {"context__org_id": "org", "context__course_id": ""},
-            "course_id must match regex",
-        ),
-        (
-            {"context__org_id": "org", "context__course_id": "course-v1:org"},
-            "course_id must match regex",
-        ),
-        (
-            {"context__org_id": "org", "context__course_id": "course-v1:org+course"},
-            "course_id must match regex",
-        ),
-        (
-            {"context__org_id": "org", "context__course_id": "course-v1:org+course+"},
-            "course_id must match regex",
-        ),
-    ],
+@settings(max_examples=1)
+@given(
+    st.builds(
+        BaseEvent, context=st.builds(BaseContextField), referer=provisional.urls()
+    )
 )
-def test_models_edx_base_event_with_invalid_content(kwargs, error):
-    """Tests that an invalid base event raises a ValidationError."""
+def test_models_edx_base_event_with_valid_content(event):
+    """Tests that a valid base event does not raise a ValidationError."""
 
-    with pytest.raises(ValidationError, match=error):
-        BaseEventFactory(**kwargs)
+    assert len(event.username) == 0 or (len(event.username) in range(2, 31, 1))
+    assert (
+        re.match(r"^course-v1:.+\+.+\+.+$", event.context.course_id)
```

Comment:
To test a regex I would prefer to see some example values it's matching and others for which it's not matching. WDYT ?

Reply:
You're right, it is something to our model. Coming in my next push